shuup/front/templates/shuup/front/macros/product_information.jinja

```diff
@@ -113,32 +113,25 @@
                             <dt>{% trans %}Available Until{% endtrans %}</dt>
                             <dd>{{ shop_product.available_until|date }}</dd>
                         {% endif %}
+                        {% set mass_unit, length_unit = get_mass_and_measurement_unit() %}
                         {% if product.width or product.height or product.depth %}
                             <dt>{% trans %}Width{% endtrans %}</dt>
-                            <dd>{{ product.width|floatformat(2) }} mm</dd>
+                            <dd>{{ product.width|floatformat(2) }} {{ length_unit }}</dd>
                             <dt>{% trans %}Height{% endtrans %}</dt>
-                            <dd>{{ product.height|floatformat(2) }} mm</dd>
+                            <dd>{{ product.height|floatformat(2) }} {{ length_unit }}</dd>
                             <dt>{% trans %}Depth{% endtrans %}</dt>
-                            <dd>{{ product.depth|floatformat(2) }} mm</dd>
+                            <dd>{{ product.depth|floatformat(2) }} {{ length_unit }}</dd>
                         {% endif %}
                         {% if product.net_weight %}
                             <dt>{% trans %}Net Weight{% endtrans %}</dt>
                             <dd>
-                                {% if product.net_weight < 1000 %}
-                                    {{ product.net_weight|floatformat(2) }} g
-                                {% else %}
-                                    {{ (product.net_weight / 1000)|floatformat(2) }} kg
-                                {% endif %}
```

Comment:
@tulimaki What do you think, do we need to special case this `g`/`kg` displaying for backwards compatibility?

Reply:
Well how this would work? You are showing product net weight on a template and that net wight have measurement unit defined by settings?

Reply:
That's why I said *special* case, e.g.

```
{% if mass_unit == "g" %}
    {# Special casing the conversion for backwards compatibility. #} 
    {% if product.net_weight < 1000 %}
        {{ product.net_weight|floatformat(2) }} g
    {% else %}
        {{ (product.net_weight / 1000)|floatformat(2) }} kg
    {% endif %}
{% else %}
    {{ product.gross_weight|floatformat(2) }} {{ mass_unit }}
{% endif %}
```

Reply:
No conversions. Keep it simple.

Reply:
No tests.